src/msw/window.cpp

```diff
@@ -4351,9 +4359,14 @@ bool wxWindowMSW::HandleEndSession(bool endSession, long logOff)
     if ( !endSession )
         return false;
```

Comment:
This branch should be amended with `gs_gotEndSession = false;` so that any subsequent new attempts to end the session work correctly.

Reply:
We could do it, but I think it's unnecessary and potentially confusing because it contradicts the claim that once `gs_gotEndSession` is set to true the application is sure to exit. Am I missing some scenario in which you think we may get `WM_ENDSESSION(TRUE)` first followed by `WM_ENDSESSION(FALSE)` later?

Reply:
Sorry, I got confused about `WM_ENDSESSION` vs `WM_QUERYENDSESSION`. You're right, sorry :(